### ISSUE TITLE
Probes marked as debris are selected in various contracts

### ISSUE THREAD

**Issue by jhuber-de** [severedsolo/CleverSat#3]
CleverSat creates contracts for probes that are marked as debris.

I launched a satellite for a contract and after completing the contract I changed the label from probe to debris. A few days later I got a contract to change the orbit of that satellite and after that a contract to move the satellite into an constellation.

**Comment by severedsolo** [comment]
Kinda deliberate, because I like to set my sats as debris, so they aren't clogging up my map view.

However, you are right, and it's not a great gameplay choice. Generally speaking, if the player has marked something as debris, it should go away.

So, before I go ahead and do something drastic and just say "no debris" - why are these probes marked as debris/why don't you want them to move? Are they out of fuel (in which case, I can put a fuel check in), or do you just not want them moved?

**Comment by jhuber-de** [comment]
Well I can see that point. Since I wont use the sats that are marked as debris I could destroy them anyway and then the problem would be solved. Thats probably the better way.

A nice way to solve the issue in general would be to tell the player in the contract description that the current contract involves a sat that is marked as debris, so that the player dont search the sat and can't find it because debris is hidden. Or to allow the user to toogle that setting on or of but I have no idea if that is even possible

Thats especially a problem if the sats for the contracts use the same launcher and thereby have the same name. But maybe I'm the only lazy person to do so...

EDIT: No the sats are fully functional

**Comment by severedsolo** [comment]
Well, to be fair the sat is highlighted in the map screen (it gets the little square around it, that contextual contracts get) but that's a fair point, and something that can be easily done. (the warning, not the toggle. Although a MM patch would easily fix the debris thing)

**Comment by jhuber-de** [comment]
Ohhhh thats what the squares are for... thats embarrassing... I'm sorry but somehow that never cought my attention. Maybe because the squares don't show up if the sat is hidden due to beeing clasified as debris, but I never tought about that they are from the sat contracts since the looked to appear randomly.

**Comment by severedsolo** [comment]
Do they not? They did for me last time I checked, but you just can't see the craft until you unhide debris.

It's not just sat contracts, any Contract Configurator contract (and a few stock ones) that require a vessel will highlight the required vessel.

**Comment by jhuber-de** [comment]
In case I didn't do anything stupid again the squares aren't there. There is this orbit of the sat drawn but that looks like an normal sat contract orbit.
Thats probably why I overlooked it.
![screenshot3](https://cloud.githubusercontent.com/assets/9028609/17190044/91467974-5444-11e6-8ec8-92c454be3d73.png)

And here you see that the green orbit is not for a new sat but for the contract.
![screenshot4](https://cloud.githubusercontent.com/assets/9028609/17190045/94e28532-5444-11e6-8130-0315c11d2129.png)

The fact that there isn't actually an contract for putting a sat around kerbin must be just something else I had overlooked.

So I think this issue could be closed since the problem was mostly stupidity on my side.

PS: I realy like how your contracts work :) you did a fantastic job. Imo thats how the stock sat contracts should be like
